rust_src/src/buffers.rs

```diff
@@ -360,4 +360,25 @@ fn set_buffer(buffer_or_name: LispObject) -> LispObject {
     buffer
 }
 
+/// No such buffer error.
+#[no_mangle]
+pub fn nsberror(spec: LispObject) {
+    if spec.is_string() {
+        let msg = "No buffer named ";
+        let mut argsarray = [
+            unsafe { make_string(msg.as_ptr() as *const c_char, msg.len() as ptrdiff_t) },
+            spec.to_raw(),
+        ];
+        xsignal!(
+            ::remacs_sys::Qerror,
+            LispObject::from(Fconcat(
+                argsarray.len() as ptrdiff_t,
+                argsarray.as_mut_ptr(),
+            ))
+        );
+    } else {
+        error!("Invalid buffer argument");
```

Comment:
My preference would be to see this if/else inverted. That way the code exits early and the rest of the function is not indented in an if.

Reply:
Ok.